test/pipelines/duplicates_filter_test.exs

```diff
@@ -0,0 +1,40 @@
+defmodule Pipelines.DuplicatesFilterTest do
+  use ExUnit.Case, async: false
+
+  @valid %{data: [%{some: "nested_data"}], id: "my_id"}
+
+  test "DuplicatesFilter prevents drops duplicate items with the same item_id value" do
+    Application.put_env(:crawly, :item_id, :id)
```

Comment:
CI returns an error here:
```
  1) test DuplicatesFilter is inactive when item_id is not set (Pipelines.DuplicatesFilterTest)
     test/pipelines/duplicates_filter_test.exs:23
     Assertion with == failed
     code:  assert Map.has_key?(state, :duplicates_filter) == false
     left:  true
     right: false
     stacktrace:
       test/pipelines/duplicates_filter_test.exs:31: (test)```

It's required to cleanup the Application state, as otherwise it will affect the second test. Please add the following setup:
```
  setup do
    on_exit(fn ->
      Application.put_env(:crawly, :item_id, nil)
    end)
  end
```